### PR TITLE
ENH: Add ITKVkFFTBackend remote module

### PR DESCRIPTION
**Pull request by tbirdso**
Changes:
- Adds [https://github.com/InsightSoftwareConsortium/ITKVkFFTBackend](https://github.com/InsightSoftwareConsortium/ITKVkFFTBackend) as a remote module
- Removes deprecated `ITK/FindOpenCL.cmake` in favor of CMake's built-in procedure for finding OpenCL, which extends the list of variables defined for referencing OpenCL and is compatible with [CMake>=3.10](https://cmake.org/cmake/help/latest/module/FindOpenCL.html).
- Removes interface libraries from IDE directory organization for compatibility with CMake<3.19

## PR Checklist
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

File: Modules/Remote/VkFFTBackend.remote.cmake [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#--   - [X] Key API features are exposed in wrapping interface
+#--   - [ ] All requirements of Levels 4,3,2,1
+#--
+#-- ## Compliance Level 4 star (Very high-quality code, perhaps small community dependance)
+#--   - [X] Meets all ITK code style standards
+#--   - [ ] No external requirements beyond those needed by ITK proper
+#--   - [ ] Builds and passes tests on all supported platforms within 1 month of each core tagged release
+#--            - [ ] Windows Shared Library Build with Visual Studio
+#--            - [ ] Mac with clang compiller
+#--            - [ ] Linux with gcc compiler
```
Comment (Leengit) [inline Modules/Remote/VkFFTBackend.remote.cmake:18]: Can we check off any of these?

Comment (tbirdso) [reply Modules/Remote/VkFFTBackend.remote.cmake:18]: Sounds reasonable to me. I will make the change. I am also chasing down CMake bugs related to building VkFFT dependencies with OpenCL.

File: Modules/Remote/VkFFTBackend.remote.cmake [hunk 2]
```diff
@@ -12,10 +12,10 @@
 #-- ## Compliance Level 4 star (Very high-quality code, perhaps small community dependance)
 #--   - [X] Meets all ITK code style standards
 #--   - [ ] No external requirements beyond those needed by ITK proper
-#--   - [ ] Builds and passes tests on all supported platforms within 1 month of each core tagged release
-#--            - [ ] Windows Shared Library Build with Visual Studio
-#--            - [ ] Mac with clang compiller
-#--            - [ ] Linux with gcc compiler
+#--   - [X] Builds and passes tests on all supported platforms within 1 month of each core tagged release
```
Comment (dzenanz) [inline Modules/Remote/VkFFTBackend.remote.cmake:15]: Squash this into first commit before merging.

**Review by dzenanz — APPROVED** [review]
Squa

**Review by Leengit — APPROVED** [review]

**Comment by tbirdso** [comment]
Seeing two different CMake errors on Windows vs Linux, not having much luck resolving either so far.

Windows (MSVC) generate error:
```
CMake Error at CMakeLists.txt:713 (get_target_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "SOURCE_DIR" is not allowed.
Call Stack (most recent call first):
  CMakeLists.txt:726 (itk_organize_targets)


CMake Error in C:/repos/ITK-build-remote-16/_deps/vulkan_lib-src/CMakeLists.txt:
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "FOLDER" is not allowed.
```
Relevant [CMakeLists.txt](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CMakeLists.txt#L713)

Linux (Ninja) compile error:
```
FAILED: Modules/Remote/VkFFTBackend/test/CMakeFiles/VkFFTBackendTestDriver.dir/itkVkComplexToComplexFFTImageFilterTest.cxx.o
/usr/bin/c++  -I/home/tom/builds/ITK-remote/Modules/ThirdParty/Eigen3/src -I/home/tom/builds/ITK-remote/Modules/ThirdParty/KWSys/src -I/home/tom/repos/ITK/Modules/ThirdParty/VNL/src/vxl/v3p/netlib -I/home/tom/repos/ITK/Modules/ThirdParty/VNL/src/vxl/vcl -I/home/tom/repos/ITK/Modules/ThirdParty/VNL/src/vxl/core -I/home/tom/builds/ITK-remote/Modules/ThirdParty/VNL/src/vxl/v3p/netlib -I/home/tom/builds/ITK-remote/Modules/ThirdParty/VNL/src/vxl/vcl -I/home/tom/builds/ITK-remote/Modules/ThirdParty/VNL/src/vxl/core -I/home/tom/builds/ITK-remote/Modules/Core/Common -I/home/tom/repos/ITK/Modules/Core/Common/include -I/home/tom/repos/ITK/Modules/Filtering/ImageFilterBase/include -I/home/tom/repos/ITK/Modules/Filtering/ImageCompose/include -I/home/tom/repos/ITK/Modules/Core/ImageAdaptors/include -I/home/tom/builds/ITK-remote/Modules/ThirdParty/Netlib -I/home/tom/repos/ITK/Modules/Numerics/Statistics/include -I/home/tom/repos/ITK/Modules/Core/Transform/include -I/home/tom/repos/ITK/Modules/Core/ImageFunction/include -I/home/tom/repos/ITK/Modules/Filtering/ImageGrid/include -I/home/tom/builds/ITK-remote/Modules/ThirdParty/ZLIB/src -I/home/tom/builds/ITK-remote/Modules/ThirdParty/ZLIB/src/itkzlib-ng -I/home/tom/repos/ITK/Modules/ThirdParty/ZLIB/src -I/home/tom/builds/ITK-remote/Modules/ThirdParty/MetaIO/src/MetaIO/src -I/home/tom/repos/ITK/Modules/ThirdParty/MetaIO/src/MetaIO/src -I/home/tom/repos/ITK/Modules/Core/SpatialObjects/include -I/home/tom/repos/ITK/Modules/Filtering/ImageStatistics/include -I/home/tom/repos/ITK/Modules/Filtering/Path/include -I/home/tom/repos/ITK/Modules/Filtering/ImageIntensity/include -I/home/tom/repos/ITK/Modules/ThirdParty/DoubleConversion/src -I/home/tom/builds/ITK-remote/Modules/ThirdParty/DoubleConversion/src/double-conversion -I/home/tom/builds/ITK-remote/Modules/IO/ImageBase -I/home/tom/repos/ITK/Modules/IO/ImageBase/include -I/home/tom/repos/ITK/Modules/Core/TestKernel/include -I/home/tom/repos/ITK/Modules/Filtering/FFT/include -I/home/tom/repos/ITK/Modules/Filtering/ImageSources/include -I/home/tom/repos/ITK/Modules/Remote/VkFFTBackend/include -I/home/tom/repos/ITK/Modules/ThirdParty/VNL/src/vxl/core/vnl/algo -I/home/tom/repos/ITK/Modules/ThirdParty/VNL/src/vxl/core/vnl -I/home/tom/builds/ITK-remote/Modules/ThirdParty/VNL/src/vxl/core/vnl -I/home/tom/repos/ITK/Modules/ThirdParty/ZLIB/src/itkzlib-ng -isystem /home/tom/repos/ITK/Modules/ThirdParty/Eigen3/src/itkeigen/.. -mtune=generic -march=corei7 -Wall -Wcast-align -Wdisabled-optimization -Wextra -Wformat=2 -Winvalid-pch -Wno-format-nonliteral -Wpointer-arith -Wshadow -Wunused -Wwrite-strings -funit-at-a-time -Wno-strict-overflow -Wno-deprecated -Wno-invalid-offsetof -Woverloaded-virtual -Wstrict-null-sentinel  -msse2 -O3 -DNDEBUG -fPIE -std=c++14 -MD -MT Modules/Remote/VkFFTBackend/test/CMakeFiles/VkFFTBackendTestDriver.dir/itkVkComplexToComplexFFTImageFilterTest.cxx.o -MF Modules/Remote/VkFFTBackend/test/CMakeFiles/VkFFTBackendTestDriver.dir/itkVkComplexToComplexFFTImageFilterTest.cxx.o.d -o Modules/Remote/VkFFTBackend/test/CMakeFiles/VkFFTBackendTestDriver.dir/itkVkComplexToComplexFFTImageFilterTest.cxx.o -c /home/tom/repos/ITK/Modules/Remote/VkFFTBackend/test/itkVkComplexToComplexFFTImageFilterTest.cxx
In file included from /home/tom/repos/ITK/Modules/Remote/VkFFTBackend/include/itkVkComplexToComplexFFTImageFilter.h:24,
                 from /home/tom/repos/ITK/Modules/Remote/VkFFTBackend/test/itkVkComplexToComplexFFTImageFilterTest.cxx:19:
/home/tom/repos/ITK/Modules/Remote/VkFFTBackend/include/itkVkCommon.h:24:10: fatal error: vkFFT.h: No such file or directory
   24 | #include "vkFFT.h"
      |          ^~~~~~~~~
compilation terminated.
```
Headers should be included at [ITKVkFFTBackend/CMakeLists.txt:61](https://github.com/InsightSoftwareConsortium/ITKVkFFTBackend/blob/master/CMakeLists.txt#L61)


Any thoughts?

**Comment by tbirdso** [comment]
@Leengit @thewtex @dzenanz Ping for re-review at your convenience

Also adding @jcfr for discussion of CMake changes

**Review by Leengit — COMMENTED** [review]
I know nothing about when `${type}` would be `"INTERFACE_LIBRARY"` so I am no help with that code.  Otherwise looks good.

**Comment by tbirdso** [comment]
For context, [VkFFT](https://github.com/DTolm/VkFFT) generates two CMake "INTERFACE_LIBRARY" targets, [`VkFFT`](https://github.com/DTolm/VkFFT/blob/master/CMakeLists.txt#L125) and [`half`](https://github.com/DTolm/VkFFT/blob/master/CMakeLists.txt#L129)

CMake interface libraries are discussed [here](https://cmake.org/cmake/help/latest/command/add_library.html#interface-libraries)

**Review by thewtex — APPROVED** [review]
:taco: :tada:

**Pull request merged**